src/main/java/org/jfrog/hudson/util/ExtractorUtils.java

```diff
@@ -195,6 +190,27 @@ public static ArtifactoryClientConfiguration getArtifactoryClientConfiguration(M
         return configuration;
     }
 
+    private static void setPublisherResolverInfo(Map<String, String> env, Run build, ArtifactoryClientConfiguration configuration,
+                                                 PublisherContext publisherContext, ResolverContext resolverContext, BuildInfo pipelineBuildInfo) throws IOException {
+        String buildName = BuildUniqueIdentifierHelper.getBuildName(build);
+        String buildNumber = BuildUniqueIdentifierHelper.getBuildNumber(build);
+        if (pipelineBuildInfo != null) {
+            buildName = pipelineBuildInfo.getName();
+            buildNumber = pipelineBuildInfo.getNumber();
+        }
+        configuration.info.setBuildName(buildName);
```

Comment:
The ```configuration.info.setBuildName``` is set twice - once here, and also inside the ```setPublisherInfo``` method, which is invoked from this method.
I addition, in the ```setPublisherInfo``` method, the ```context.getCustomBuildName()``` is taken into account, while here it does not. This means that upon resolution, the custom build name configured is not used.

Reply:
The `configuration.info.setBuildName` may happen a second time from the `setPublisherInfo` only in Maven, Gradle and Ivy freestyle builds. The custom build name we set in other builds doesn't affect this `context.getCustomBuildName()`.